components/mpas-ocean/src/shared/mpas_ocn_tendency.F

```diff
@@ -417,6 +422,15 @@ subroutine ocn_tend_vel(tendPool, statePool, forcingPool, &
                                thermExpCoeff, salineContractCoeff, &
                                tendVel, err)
 
+      ! Compute ssh additions due to self-attraction and loading
+      if(mpas_is_alarm_ringing(domain % clock, 'salComputeAlarm', ierr=err)) then 
+#ifdef MPAS_DEBUG
+          call mpas_log_write('       Computing SAL')
+#endif
+          call ocn_compute_self_attraction_loading(domain, forcingPool, dminfo, ssh, err)
+          call mpas_reset_clock_alarm(domain % clock, 'salComputeAlarm', ierr=err)
+      endif
```

Comment:
after you've computed ssh_sal, you need to update it on the device for later use, so need to add at the end of this if block
!$acc update device (ssh_sal)